test/e2e/mcd_test.go

```diff
@@ -1122,3 +1122,197 @@ func setupForInternalImageRegistryPullSecretTest(t *testing.T, cs *framework.Cli
+
+		err := helpers.WaitForNodeReady(t, cs, node)
+		if err != nil {
+			t.Logf("Node %s is not ready: %v.", node.Name, err)
+			return
+		}
+
+		// Uninstall the package
+		uninstallCmd := []string{
+			"chroot", "/rootfs", "rpm-ostree", "uninstall", "aeskeyfind",
+		}
+
+		_, err = helpers.ExecCmdOnNodeWithError(cs, node, uninstallCmd...)
+		if err != nil {
+			t.Logf("Failed to uninstall RPM package on node %s: %v", node.Name, err)
+			return
+		}
+		t.Log("RPM package uninstalled successfully")
+
+		t.Logf("Rebooting node %s to apply the layered package", node.Name)
+
+		rebootCmd := []string{
+			"chroot", "/rootfs", "sudo", "systemctl", "reboot",
+		}
+
+		_, err = helpers.ExecCmdOnNodeWithError(cs, node, rebootCmd...)
+		if err != nil {
+			t.Logf("Failed to reboot node %s: %v", node.Name, err)
+			return
+		}
+
+		err = helpers.WaitForNodeReady(t, cs, node)
+		if err != nil {
+			t.Logf("Node %s is not ready: %v.", node.Name, err)
+			return
+		}
+		t.Logf("Node %s rebooted successfully and rpm package is removed", node.Name)
+	}()
+
+	t.Log("Waiting for the configuration to be applied to the pool")
+	helpers.WaitForConfigAndPoolComplete(t, cs, "worker", mc.Name)
+	t.Log("Configuration applied successfully")
+
+	unsupportedPackage := "/rootfs/var/tmp/aeskeyfind.rpm"
+
+	t.Logf("Checking if the unsupported package exists on node %s", node.Name)
+	fileCheckCmd := []string{
+		"ls", "-l", "/rootfs/var/tmp", unsupportedPackage,
+	}
+	_, err := helpers.ExecCmdOnNodeWithError(cs, node, fileCheckCmd...)
+	require.NoError(t, err, "Unsupported package %s not found on node %s: %v", unsupportedPackage, node.Name, err)
+	t.Logf("Unsupported package %s found on node %s", unsupportedPackage, node.Name)
+
+	installCmd := []string{
+		"chroot", "/rootfs", "rpm-ostree", "install", "/var/tmp/aeskeyfind.rpm",
+	}
+
+	t.Logf("Executing rpm-ostree install command on node %s", node.Name)
+	out, err := helpers.ExecCmdOnNodeWithError(cs, node, installCmd...)
+	require.NoError(t, err, "Failed to install unsupported package on node %s: %v", node.Name, err)
+	t.Logf("Output from rpm-ostree install: %s", out)
+
+	t.Logf("Rebooting node %s to apply the layered package", node.Name)
+
+	rebootCmd := []string{
+		"chroot", "/rootfs", "sudo", "systemctl", "reboot",
+	}
+
+	_, err = helpers.ExecCmdOnNodeWithError(cs, node, rebootCmd...)
+	require.NoError(t, err, "Failed to reboot node %s: %v", node.Name, err)
+	t.Logf("Reboot command issued successfully")
+
+	t.Logf("Waiting for node %s to be ready after reboot", node.Name)
+	require.NoError(t, helpers.WaitForNodeReady(t, cs, node))
+
+	// Wait for the MCD to detect the change and update metrics
+	t.Logf("Waiting for MCD metrics to reflect the installed package")
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+	defer cancel()
+
+	err = wait.PollUntilContextTimeout(ctx, 30*time.Second, 10*time.Minute, true, func(ctx context.Context) (done bool, err error) {
+		cmd := []string{
```

Comment:
thought (non-blocking): I vaguely recall a similar function to this elsewhere within our codebase. In the future, it might be useful to make a more general-purpose version of this code.